client/plots/controls.term1.js

```diff
@@ -146,9 +146,15 @@ function setRenderers(self) {
 				throw 'unknown term type'
 		}
 		if (!self.pill) self.setPill()
+		const isDictTerm = isDictionaryType(plot.term.term.type)
+		// assume that dictionary tw are complete after fillTermWrapper(),
+		// whereas non-dictionary terms may require additional term/q properties
+		// to be added when the pill edit UI is opened, so term/q must not be frozen
```

Comment:
@xzhou82 An assumption is made here that dictionary terms will not need new properties to be added when the pill edit menu is opened, whereas non-dict terms like geneExpression may require new term key-values to be set when rendering the edit menu, so `term`/`q` must not be frozen only for non-dict terms.